### PR TITLE
Add actionable platform example

### PR DESCRIPTION
**Pull request by akenyon**
Simple example demonstrating use of actionable platforms. Gives an intro to the platforms and movement controllers currently available in Stone Soup.

### CODE REVIEW

File: docs/examples/sensormanagement/actionable_platform_example.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+"""
+Actionable Platforms
+====================
+This example demonstrates the management of actionable platforms in Stone Soup.
+"""
+
+# %%
+# Platforms in Stone Soup
+# -----------------------
+# In Stone Soup, instances of :class:`~.Platform` are objects to which one or more sensors can be mounted, and provide a means of controlling the position of mounted sensors.
```
Comment (csherman-dstl) [inline docs/examples/sensormanagement/actionable_platform_example.py:12]:
```suggestion
# In Stone Soup, instances of the :class:`~.Platform` class are objects to which one or more sensors can be mounted. These provide a means of controlling the position of mounted sensors.
```

File: docs/examples/sensormanagement/actionable_platform_example.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+                                                    action_mapping=(0,1)),
+    sensors=[sensor])
+
+
+# %%
+# Creating a Predictor and Updater
+# --------------------------------
+# Next we create some of the standard Stone Soup components required for tracking: a predictor, which in this case creates an initial estimate of the target at each timestep according to a linear transition model, and an updater, which updates our inital estimate based on our sensor's measurements.
+# 
+# As we are working with a particle filter, we also include a resampler, which occasionally regenerates particles according to their weight/likelihood, with those particles with higher weights being preserved and replicated. A drawback of this approach is particle impoverishment, whereby repeatedly resampling higher weight particles results in a lower diversity of samples. We therefore include a regulariser to mitigate sample impoverishment by slightly moving resampled particles according to a Gaussian kernel if an acceptance probability is met.
```
Comment (csherman-dstl) [inline docs/examples/sensormanagement/actionable_platform_example.py:110]:
```suggestion
# As we are working with a particle filter, we also include a resampler, which occasionally regenerates particles according to their weight/likelihood. The particles with higher weights are preserved and replicated. A drawback of this approach is particle impoverishment, whereby repeatedly resampling higher weight particles results in a lower diversity of samples. We therefore include a regulariser to mitigate sample impoverishment by slightly moving resampled particles according to a Gaussian kernel if an acceptance probability is met.
```

File: docs/examples/sensormanagement/actionable_platform_example.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+                                                                     ConstantVelocity(0.1)]))
+updater = ParticleUpdater(sensor.measurement_model,
+                          resampler=resampler,
+                          regulariser=regulariser)
+
+
+# %%
+# Creating a Sensor Manager
+# -------------------------
+# Now we create a sensor manager, giving it our sensor and platform, and a reward function, in this case the ExpectedKLDivergence reward function, which chooses actions based on the information gained by taking that action.
```
Comment (csherman-dstl) [inline docs/examples/sensormanagement/actionable_platform_example.py:130]:
```suggestion
# Now we create a sensor manager, giving it our sensor and platform, and a reward function. In this case the ExpectedKLDivergence reward function is used, which chooses actions based on the information gained by taking that action.
```

**Review by csherman-dstl — COMMENTED** [review]
Some minor readability suggestions, the other suggestion would be to try and separate each  line to not exceed character limit.

**Review by csherman-dstl — APPROVED** [review]

File: docs/examples/sensormanagement/actionable_platform_example.py [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+updater = ParticleUpdater(sensor.measurement_model,
+                          resampler=resampler,
+                          regulariser=regulariser)
+
+
+# %%
+# Creating a Sensor Manager
+# -------------------------
+# Now we create a sensor manager, giving it our sensor and platform, and a reward function. In
+# this case the ExpectedKLDivergence reward function is used, which chooses actions based on the
```
Comment (nperree-dstl) [inline docs/examples/sensormanagement/actionable_platform_example.py:166]:
```suggestion
# this case the :class:`~.ExpectedKLDivergence' reward function is used, which chooses actions based on the
```

File: docs/examples/sensormanagement/actionable_platform_example.py [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+# --------
+# As we can see, the actionable platform is able to follow the target as it moves across the
+# action space, keeping it within the sensor's range.
+
+
+import plotly.graph_objects as go
+from stonesoup.functions import pol2cart
+
+plotter.plot_sensors(platform)
+plotter.plot_tracks(track, mapping=(0, 2))
```
Comment (nperree-dstl) [inline docs/examples/sensormanagement/actionable_platform_example.py:277]:
```suggestion
plotter.plot_tracks(track, mapping=(0, 2), particle=True)
```
Might be nice to plot the particles to show how the particle state changes over time

**Review by nperree-dstl — CHANGES_REQUESTED** [review]

File: docs/examples/sensormanagement/actionable_platform_example.py [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+# This was a simple example demonstrating how an actionable platform can be used to track a moving
+# target.
+#
+# There are many other scenarios to which actionable platforms could be applied, and the number of
+# possible behaviours and applications will only increase as additional classes are introduced.
+# Stone Soup also makes it easy to implement additional functionality on your own, for example by
+# experimenting with custom reward functions and/or movement controllers.
+#
+# To see the latest developments for actionable platforms, you can refer to the
+# `Stone Soup docs <https://stackoverflow.com/>`_.
```
[the remark this answers is not in the document]
Comment (akenyon) [reply docs/examples/sensormanagement/actionable_platform_example.py]: Added docs for movable actions and reference both in example.

File: docs/examples/sensormanagement/actionable_platform_example.py (hunk 5 above)
Comment (akenyon) [reply docs/examples/sensormanagement/actionable_platform_example.py]: Just tried this but after the first couple of steps the particle cloud was too small to be visible, so I haven't included for now.

**Review by nperree-dstl — APPROVED** [review]

**Pull request merged**